sweep/fee_function.go

```diff
@@ -86,11 +86,14 @@ type LinearFeeFunction struct {
 	currentFeeRate chainfee.SatPerKWeight
 
 	// width is the number of blocks between the starting block height
-	// and the deadline block height.
+	// and the deadline block height minus one.
+	//
+	// NOTE: We do minus one here because we want to max out the budget
+	// before the deadline height is reached.
 	width uint32
 
-	// position is the number of blocks between the starting block height
-	// and the current block height.
+	// position is the fee function's current position, given a width of w,
+	// a valid position should lie in range [0, w-1].
```

Comment:
Width now maxes out at `deadline - 1`, so we should allow position to be in [0, width].

```suggestion
	// position is the fee function's current position, given a width of w,
	// a valid position should lie in range [0, w].
```

Reply:
fixed!